src/smt/smt_engine_state.h

```diff
@@ -0,0 +1,240 @@
+   * to simplify() is issued, a scope is pushed, etc.
+   */
+  bool isFullyInited() const;
+  /**
+   * Return true if the SmtEngine is fully initialized and there are no
+   * pending pops.
+   */
+  bool isFullyReady() const;
+  /**
+   * Return true if a notifyCheckSat call has been made, e.g. a query has been
+   * issued to the SmtEngine.
+   */
+  bool isQueryMade() const;
+  /** Return the user context level.  */
+  size_t getNumUserLevels() const;
+  /** Get the status of the last check-sat */
+  Result getStatus() const;
+  /** Get the SMT mode we are in */
+  SmtMode getMode() const;
+  /** return the input name (if any) */
+  std::string getFilename() const;
+  //---------------------------- end queries
+
+ private:
+  /** Pushes the user and SAT contexts */
+  void push();
+  /** Pops the user and SAT contexts */
+  void pop();
+  /** Pops the user and SAT contexts to the given level */
+  void popto(int toLevel);
+  /**
+   * Internal push, which processes any pending pops, and pushes (if in
+   * incremental mode).
+   */
+  void internalPush();
+  /**
+   * Internal pop. If immediate is true, this processes any pending pops, and
+   * pops (if in incremental mode). Otherwise, it increments the pending pop
+   * counter and does nothing.
+   */
+  void internalPop(bool immediate = false);
+  /** Reference to the SmtEngine */
+  SmtEngine& d_smt;
+  /** Expr manager context */
+  std::unique_ptr<context::Context> d_context;
+  /** User level context */
+  std::unique_ptr<context::UserContext> d_userContext;
+  /** The context levels of user pushes */
+  std::vector<int> d_userLevels;
+
+  /**
+   * Number of internal pops that have been deferred.
+   */
+  unsigned d_pendingPops;
+
+  /**
+   * Whether or not this SmtEngine is fully initialized (post-construction).
+   * This post-construction initialization is automatically triggered by the
+   * use of the SmtEngine; e.g. when the first formula is asserted, a call
+   * to simplify() is issued, a scope is pushed, etc.
+   */
+  bool d_fullyInited;
+
+  /**
+   * Whether or not a checkEntailed() or checkSatisfiability() has already been
+   * made through this SmtEngine.  If true, and incrementalSolving is false,
+   * then attempting an additional checkEntailed() or checkSat() will fail with
+   * a ModalException.
+   */
+  bool d_queryMade;
+
+  /**
+   * Internal status flag to indicate whether we've sent a theory
+   * presolve() notification and need to match it with a postsolve().
+   */
+  bool d_needPostsolve;
+
+  /**
+   * Most recent result of last checkSatisfiability/checkEntailed or
+   * (set-info :status).
+   */
+  Result d_status;
+
+  /**
+   * The expected status of the next satisfiability check.
+   */
+  Result d_expectedStatus;
+
+  /** The SMT mode, for details see enumeration above. */
+  SmtMode d_smtMode;
+
+  /**
+   * The input file name (if any) or the name set through setInfo (if any)
```

Comment:
```suggestion
   * The input file name (if any)
```

Reply:
setInfo still sets this field actually.

Reply:
Right, but the comment suggests that `setInfo` is a method in this class which it is not as far as I can tell. Maybe write `or the name set through (set-info :filename ...)`?